elytron-common/src/main/java/org/wildfly/extension/elytron/common/ElytronCommonCapabilities.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.extension.elytron.common;
+
+import java.security.KeyStore;
+import java.security.Provider;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.wildfly.extension.elytron.common.capabilities.PrincipalTransformer;
+import org.wildfly.security.auth.server.RealmMapper;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.x500.cert.acme.AcmeAccount;
+import org.wildfly.security.x500.cert.acme.CertificateAuthority;
+
+/**
+ * The capabilities provided by and required by Elytron.
+ *
+ * @implNote It is a deliberate decision that this class is not public, by using capability definitions it should be possible to
+ * completely remove this subsystem and allow another to provide all the capabilities - allowing references to this class would
+ * not allow complete removal.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ * @author <a href="mailto:carodrig@redhat.com">Cameron Rodriguez</a>
+ */
+public abstract class ElytronCommonCapabilities {
+
+    protected static final String CAPABILITY_BASE = "org.wildfly.security.";
+
+    public static final String CERTIFICATE_AUTHORITY_ACCOUNT_CAPABILITY = CAPABILITY_BASE + "certificate-authority-account";
+
+    public static final RuntimeCapability<Void> CERTIFICATE_AUTHORITY_ACCOUNT_RUNTIME_CAPABILITY =  RuntimeCapability
+            .Builder.of(CERTIFICATE_AUTHORITY_ACCOUNT_CAPABILITY, true, AcmeAccount.class)
+            .build();
+
+    public static final String CERTIFICATE_AUTHORITY_CAPABILITY = CAPABILITY_BASE + "certificate-authority";
+
+    public static final RuntimeCapability<Void> CERTIFICATE_AUTHORITY_RUNTIME_CAPABILITY =  RuntimeCapability
+            .Builder.of(CERTIFICATE_AUTHORITY_CAPABILITY, true, CertificateAuthority.class)
+            .build();
+
+    public static final String ELYTRON_CAPABILITY = CAPABILITY_BASE + "elytron";
```

Comment:
@fjuma As I was working on the TLS subsystem, I went over the uses of this capability again, and I don't think it should be in `elytron-common`. There are a few mentions in the basic add and remove handlers (ex. [ElytronCommonBaseAddHandler](https://github.com/cam-rod/wildfly-core/blob/782f37e11f8335e2b4c9813d5db1cfe6e0e764aa/elytron-common/src/main/java/org/wildfly/extension/elytron/common/ElytronCommonBaseAddHandler.java#L83)), but I'm thinking that instead I should be passing the capability of the implementing subsystem as an argument (ex. `ELYTRON_TLS_CAPABILITY`). Is this correct?

Reply:
@cam-rod Yes, for this particular capability, I think it makes sense to not have this in `elytron-common`.